File: apps/platform/md/server/api/models/store.py
"""
Store 数据模型定义
多产品线商城模块的数据模型
"""

from pydantic import BaseModel, Field
from typing import Optional, List, Any
from datetime import datetime


class StoreCollection(BaseModel):
    """商城产品集合"""
    id: Optional[int] = None
    collection_key: str  # 集合标识键，如 'sillyclaw', 'electronics'
    name_zh: str  # 中文名称
    name_en: Optional[str] = None  # 英文名称
    description: Optional[str] = None  # 描述
    logo_url: Optional[str] = None  # Logo URL
    theme_config: Optional[dict] = None  # 主题配置 JSON
    is_active: bool = True  # 是否激活
    sort_order: int = 0  # 排序
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class StoreProduct(BaseModel):
    """商城产品"""
    id: Optional[int] = None
    collection_id: int  # 所属集合ID
    product_key: str  # 产品标识键
    name_zh: str  # 中文名称
    name_en: Optional[str] = None  # 英文名称
    description_zh: Optional[str] = None  # 中文描述
    description_en: Optional[str] = None  # 英文描述
    image_url: Optional[str] = None  # 主图
    gallery: Optional[List[str]] = []  # 图片列表
    price: float  # 价格
    original_price: Optional[float] = None  # 原价（用于折扣显示）
    stock_count: int = -1  # 库存，-1表示无限
    is_active: bool = True  # 是否上架
    specifications: Optional[dict] = None  # 技术规格 JSON
    sort_order: int = 0  # 排序
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class StoreProductDetail(StoreProduct):
    """产品详情（含集合信息）"""
    collection_name: Optional[str] = None


class StoreCartItem(BaseModel):
    """购物车项"""
    id: Optional[int] = None
    user_id: int
    collection_id: int
    product_id: int
    quantity: int = 1
    product_key: Optional[str] = None
    product_name: Optional[str] = None
    product_image: Optional[str] = None
    price: Optional[float] = None
    total_price: Optional[float] = None
    stock_available: bool = True

    class Config:
        from_attributes = True


class StoreOrder(BaseModel):
    """商城订单"""
    id: Optional[int] = None
    order_no: str  # 订单号
    user_id: int
    collection_id: int
    total_amount: float  # 总金额
    status: str = "pending"  # pending, paid, shipped, completed, cancelled
    payment_method: Optional[str] = None  # wechat, alipay
    payment_no: Optional[str] = None  # 支付平台交易号
    shipping_name: Optional[str] = None  # 收货人
    shipping_phone: Optional[str] = None  # 联系电话
    shipping_address: Optional[str] = None  # 收货地址
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class StoreOrderItem(BaseModel):
    """订单项"""
    id: Optional[int] = None
    order_id: int
    product_id: int
    product_name: str
    quantity: int
    unit_price: float
    subtotal: float

    class Config:
        from_attributes = True


# ==================== Request/Response Models ====================

class CartAddRequest(BaseModel):
    """添加到购物车请求"""
    product_id: int
    quantity: int = Field(default=1, ge=1, le=99)


class CartUpdateRequest(BaseModel):
    """更新购物车请求"""
    quantity: int = Field(..., ge=1, le=99)


class CreateOrderRequest(BaseModel):
    """创建订单请求"""
    shipping_name: str = Field(..., min_length=1, max_length=100)
    shipping_phone: str = Field(..., min_length=1, max_length=20)
    shipping_address: str = Field(..., min_length=1, max_length=500)


class PaymentRequest(BaseModel):
    """支付请求"""
    payment_method: str = Field(..., description="支付方式: wechat, alipay")


# ==================== Admin Request Models ====================

class StoreCollectionCreate(BaseModel):
    """创建商城集合请求"""
    collection_key: Optional[str] = Field(None, max_length=50, description="集合标识键，不提供则自动生成")
    name_zh: str = Field(..., min_length=1, max_length=100, description="中文名称")
    name_en: Optional[str] = Field(None, max_length=100, description="英文名称")
    description: Optional[str] = Field(None, max_length=500, description="描述")
    logo_url: Optional[str] = Field(None, max_length=500, description="Logo URL")
    theme_config: Optional[dict] = None
    is_active: bool = True
    sort_order: int = 0


class StoreCollectionUpdate(BaseModel):
    """更新商城集合请求"""
    collection_key: Optional[str] = Field(None, max_length=50)
    name_zh: Optional[str] = Field(None, min_length=1, max_length=100)
    name_en: Optional[str] = Field(None, max_length=100)
    description: Optional[str] = Field(None, max_length=500)
    logo_url: Optional[str] = Field(None, max_length=500)
    theme_config: Optional[dict] = None
    is_active: Optional[bool] = None
    sort_order: Optional[int] = None


class StoreProductCreate(BaseModel):
    """创建商城产品请求"""
    collection_id: int = Field(..., description="所属集合ID")
    product_key: Optional[str] = Field(None, max_length=50, description="产品标识键，不提供则自动生成")
    name_zh: str = Field(..., min_length=1, max_length=200, description="中文名称")
    name_en: Optional[str] = Field(None, max_length=200, description="英文名称")
    description_zh: Optional[str] = Field(None, max_length=2000, description="中文描述")
    description_en: Optional[str] = Field(None, max_length=2000, description="英文描述")
    image_url: Optional[str] = Field(None, max_length=500, description="主图URL")
    gallery: Optional[List[str]] = Field(default_factory=list, description="图片列表")
    price: float = Field(..., ge=0, description="价格")
    original_price: Optional[float] = Field(None, ge=0, description="原价")
    stock_count: int = Field(-1, ge=-1, description="库存，-1表示无限")
    specifications: Optional[dict] = None
    is_active: bool = True
    sort_order: int = 0


class StoreProductUpdate(BaseModel):
    """更新商城产品请求"""
    collection_id: Optional[int] = None
    product_key: Optional[str] = Field(None, max_length=50)
    name_zh: Optional[str] = Field(None, min_length=1, max_length=200)
    name_en: Optional[str] = Field(None, max_length=200)
    description_zh: Optional[str] = Field(None, max_length=2000)
    description_en: Optional[str] = Field(None, max_length=2000)
    image_url: Optional[str] = Field(None, max_length=500)
    gallery: Optional[List[str]] = None
    price: Optional[float] = Field(None, ge=0)
    original_price: Optional[float] = Field(None, ge=0)
    stock_count: Optional[int] = Field(None, ge=-1)
    specifications: Optional[dict] = None
    is_active: Optional[bool] = None
    sort_order: Optional[int] = None


class OrderStatusUpdate(BaseModel):
    """更新订单状态请求"""
    status: str = Field(..., description="新状态: pending, paid, shipped, completed, cancelled")
    reason: Optional[str] = Field(None, max_length=500, description="状态变更原因")


# ==================== Admin Response Models ====================

class StoreProductListResponse(BaseModel):
    """产品列表响应（含分页）"""
    items: List[StoreProductDetail]
    total: int
    page: int
    page_size: int
    total_pages: int


class StoreCollectionListResponse(BaseModel):
    """集合列表响应（含分页）"""
    items: List[StoreCollection]
    total: int
    page: int
    page_size: int
    total_pages: int


class StoreOrderDetail(BaseModel):
    """订单详情（含订单项）"""
    order: StoreOrder
    items: List[StoreOrderItem]


class AdminStoreStats(BaseModel):
    """商城统计信息"""
    total_collections: int
    active_collections: int
    total_products: int
    active_products: int
    total_orders: int
    pending_orders: int
    total_revenue: float

File: apps/platform/md/admin-v2/src/api/store.ts
import apiClient from './client';
import type { ApiResponse, PaginatedResponse, PaginationParams } from '@/types';

// ---- Collection Types ----

export interface Collection {
  id: number;
  collection_key: string;
  name_zh: string;
  name_en?: string;
  description?: string;
  logo_url?: string;
  image_url?: string;
  theme_config?: Record<string, unknown>;
  sort_order: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface CollectionCreateRequest {
  name_zh: string;
  name_en?: string;
  collection_key?: string;
  description?: string;
  logo_url?: string;
  image_url?: string;
  theme_config?: Record<string, unknown>;
  sort_order?: number;
  is_active?: boolean;
}

export interface CollectionUpdateRequest
  extends Partial<CollectionCreateRequest> {}

// ---- Product Types ----

export interface Product {
  id: number;
  collection_id: number;
  product_key: string;
  name_zh: string;
  name_en?: string;
  description_zh?: string;
  description_en?: string;
  image_url?: string;
  gallery?: string[];
  price: number;
  original_price?: number;
  stock_count: number;
  sold_count: number;
  specifications?: string;
  is_active: boolean;
  sort_order: number;
  created_at: string;
  updated_at: string;
  collection_name?: string;
}

export interface ProductCreateRequest {
  collection_id?: number;
  name_zh: string;
  name_en?: string;
  product_key?: string;
  description_zh?: string;
  description_en?: string;
  image_url?: string;
  gallery?: string[];
  price: number;
  original_price?: number;
  stock_count?: number;
  specifications?: string;
  is_active?: boolean;
  sort_order?: number;
}

export interface ProductUpdateRequest
  extends Partial<ProductCreateRequest> {}

// ---- Order Types ----

export interface StoreOrder {
  id: number;
  order_no: string;
  user_id: number;
  collection_id: number;
  total_amount: number;
  status: string;
  payment_method?: string;
  payment_no?: string;
  shipping_name?: string;
  shipping_phone?: string;
  shipping_address?: string;
  created_at: string;
  updated_at?: string;
}

export interface StoreOrderItem {
  id: number;
  order_id: number;
  product_id: number;
  product_name: string;
  quantity: number;
  unit_price: number;
  subtotal: number;
}

export interface OrderDetail {
  order: StoreOrder;
  items: StoreOrderItem[];
}

export interface OrderStatusUpdateRequest {
  status: string;
  reason?: string;
}

// ---- Inventory Types ----

export interface InventoryItem {
  id: number;
  product_key: string;
  name_zh: string;
  name_en?: string;
  stock_count: number;
  sold_count: number;
  price: number;
  original_price?: number;
  is_active: boolean;
  collection_name?: string;
  collection_key?: string;
}

export interface StockLog {
  id: number;
  product_id: number;
  change_type: string;
  change_quantity: number;
  stock_before: number;
  stock_after: number;
  reference_no?: string;
  external_ref?: string;
  source: string;
  note?: string;
  operator_id?: number;
  operator_name?: string;
  sync_status: string;
  created_at: string;
}

export interface StockAdjustRequest {
  change_type: string;
  quantity: number;
  note?: string;
  reference_no?: string;
  operator_id?: number;
}

// ---- Stats Types ----

export interface StoreStats {
  total_collections: number;
  active_collections: number;
  total_products: number;
  active_products: number;
  total_orders: number;
  pending_orders: number;
  total_revenue: number;
}

// ---- Collections CRUD ----

export const getCollections = async (
  params: PaginationParams = {}
): Promise<PaginatedResponse<Collection>> => {
  const response = await apiClient.get<PaginatedResponse<Collection>>(
    '/store/admin/collections',
    { params }
  );
  return response.data;
};

export const createCollection = async (
  data: CollectionCreateRequest
): Promise<ApiResponse<Collection>> => {
  const response = await apiClient.post<ApiResponse<Collection>>(
    '/store/admin/collections',
    data
  );
  return response.data;
};

export const updateCollection = async (
  id: number,
  data: CollectionUpdateRequest
): Promise<ApiResponse<Collection>> => {
  const response = await apiClient.put<ApiResponse<Collection>>(
    `/store/admin/collections/${id}`,
    data
  );
  return response.data;
};

export const deleteCollection = async (
  id: number
): Promise<ApiResponse<null>> => {
  const response = await apiClient.delete<ApiResponse<null>>(
    `/store/admin/collections/${id}`
  );
  return response.data;
};

// ---- Products CRUD ----

export const getProducts = async (
  params: PaginationParams = {}
): Promise<PaginatedResponse<Product>> => {
  const response = await apiClient.get<PaginatedResponse<Product>>(
    '/store/admin/products',
    { params }
  );
  return response.data;
};

export const createProduct = async (
  data: ProductCreateRequest
): Promise<ApiResponse<Product>> => {
  const response = await apiClient.post<ApiResponse<Product>>(
    '/store/admin/products',
    data
  );
  return response.data;
};

export const updateProduct = async (
  id: number,
  data: ProductUpdateRequest
): Promise<ApiResponse<Product>> => {
  const response = await apiClient.put<ApiResponse<Product>>(
    `/store/admin/products/${id}`,
    data
  );
  return response.data;
};

export const deleteProduct = async (
  id: number
): Promise<ApiResponse<null>> => {
  const response = await apiClient.delete<ApiResponse<null>>(
    `/store/admin/products/${id}`
  );
  return response.data;
};

// ---- Orders ----

export const getOrders = async (
  params: PaginationParams & { status?: string; order_no?: string; collection_id?: number }
): Promise<PaginatedResponse<StoreOrder>> => {
  const response = await apiClient.get<PaginatedResponse<StoreOrder>>(
    '/store/admin/orders',
    { params }
  );
  return response.data;
};

export const getOrderDetail = async (
  orderNo: string
): Promise<ApiResponse<OrderDetail>> => {
  const response = await apiClient.get<ApiResponse<OrderDetail>>(
    `/store/admin/orders/${orderNo}/detail`
  );
  return response.data;
};

export const updateOrderStatus = async (
  orderNo: string,
  data: OrderStatusUpdateRequest
): Promise<ApiResponse<StoreOrder>> => {
  const response = await apiClient.put<ApiResponse<StoreOrder>>(
    `/store/admin/orders/${orderNo}/status`,
    data
  );
  return response.data;
};

// ---- Inventory ----

export const getInventory = async (
  params: PaginationParams & { search?: string; low_stock_only?: boolean; collection_key?: string }
): Promise<PaginatedResponse<InventoryItem>> => {
  const response = await apiClient.get<PaginatedResponse<InventoryItem>>(
    '/store/admin/inventory',
    { params }
  );
  return response.data;
};

export const adjustStock = async (
  productId: number,
  data: StockAdjustRequest
): Promise<ApiResponse<Record<string, unknown>>> => {
  const response = await apiClient.post<ApiResponse<Record<string, unknown>>>(
    `/store/admin/products/${productId}/stock-adjust`,
    data
  );
  return response.data;
};

export const getStockLogs = async (
  productId: number,
  params: PaginationParams = {}
): Promise<PaginatedResponse<StockLog>> => {
  const response = await apiClient.get<PaginatedResponse<StockLog>>(
    `/store/admin/products/${productId}/stock-logs`,
    { params }
  );
  return response.data;
};

// ---- Stats ----

export const getStoreStats = async (): Promise<ApiResponse<StoreStats>> => {
  const response = await apiClient.get<ApiResponse<StoreStats>>(
    '/store/admin/stats'
  );
  return response.data;
};
